File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr=0
        max_sum=float('-inf')
        for i in range(len(nums)):
            curr+=nums[i]
            max_sum=max(curr,max_sum)
            if curr<0:
                curr=0
        return max_sum

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int curr=0;
        int max_sum=INT_MIN;
        for (int i =0; i<nums.size();i++){
            curr+=nums[i];
            max_sum=max(curr,max_sum);
            if (curr<0){
               curr=0;
            }
        }
        return max_sum;
    }
};